src/ibex_bluesky_core/callbacks/__init__.py

```diff
@@ -1 +1,203 @@
+            raise ValueError(
+                "Fit has been requested to show on plot without a fitting method or callback."
+            )
+        if (add_peak_stats or add_fit_cb or add_plot_cb) and (x is None or y is None):
+            raise ValueError(
+                "X and/or Y not specified when trying to add a plot, fit or peak stats."
+            )
+
+        self.subs = []
+        if add_human_readable_file_cb:
+            combined_hr_fields = measured_fields + fields_for_hr_file
+            if not combined_hr_fields:
+                raise ValueError("No fields specified for the human-readable file")
+            self.subs.append(
+                HumanReadableFileCallback(
+                    fields=combined_hr_fields,
+                    output_dir=Path(human_readable_file_output_dir)
+                    if human_readable_file_output_dir
+                    else DEFAULT_PATH,
+                ),
+            )
+
+        if add_table_cb:
+            combined_lt_fields = measured_fields + fields_for_live_table
+            if not combined_lt_fields:
+                raise ValueError("No fields specified for the live table")
+            self.subs.append(
+                LiveTable(combined_lt_fields),
+            )
+
+        if add_peak_stats:
+            self._peak_stats = PeakStats(x=x, y=y)
+            self.subs.append(self._peak_stats)
+
+        if (add_plot_cb or show_fit_on_plot) and not ax:
+            logger.debug("No axis provided, creating a new one")
+            fig, ax, exc, result = None, None, None, None
+            done_event = threading.Event()
+
+            class _Cb(QtAwareCallback):
+                def start(self, doc: RunStart) -> None:
+                    nonlocal result, exc, fig, ax
+                    try:
+                        plt.close("all")
+                        fig, ax = plt.subplots()
+                    finally:
+                        done_event.set()
+
+            cb = _Cb()
+            cb("start", {"time": 0, "uid": ""})
+            done_event.wait(10.0)
+
+        if add_fit_cb:
+            if fit is None:
+                raise ValueError("fit method must be specified if add_fit_cb is True")
+            self._live_fit = LiveFit(fit, y=y, x=x, yerr=yerr)  # pyright: ignore reportArgumentType
+            if show_fit_on_plot:
+                self.subs.append(LiveFitPlot(livefit=self._live_fit, ax=ax))
+
+        if add_plot_cb or show_fit_on_plot:
+            self.subs.append(
+                LivePlot(
+                    y=y,  # pyright: ignore reportArgumentType
+                    x=x,
+                    marker="x",
+                    linestyle="none",
+                    ax=ax,
+                    yerr=yerr,
+                )
+            )
+
+    @property
+    def live_fit(self) -> LiveFit | None:
+        """The live fit object containing fitting results."""
+        return self._live_fit
```

Comment:
I think for pyright we want this to be `-> LiveFit` and inside the implementation throw a `ValueError` or something if it's None.

Otherwise people have to do:

```
the_fit = icc.live_fit
assert the_fit is not None
print(the_fit.fit_report())
```
to placate pyright downstream